lib/std/http/Client.zig

```diff
@@ -627,6 +633,23 @@ pub const Request = struct {
         try buffered.flush();
     }
 
+    fn writeUri(uri_writer: anytype, uri: std.Uri, raw_uri: bool, proxied: bool) !void {
+        if (proxied) {
+            // proxied connections require the full uri
+            if (raw_uri) {
+                try uri_writer.print("{+/r}", .{uri});
+            } else {
+                try uri_writer.print("{+/}", .{uri});
+            }
+        } else {
+            if (raw_uri) {
+                try uri_writer.print("{/r}", .{uri});
+            } else {
+                try uri_writer.print("{/}", .{uri});
+            }
+        }
+    }
```

Comment:
Does this really need to be it's own function? It will only ever be used in one place.

Reply:
No, but the nested if/else stuff looked messy. I'll inline